pkg/apis/configuration/validation/virtualserver_test.go

```diff
@@ -69,6 +69,131 @@ func TestValidateVirtualServer(t *testing.T) {
 	}
 }
 
+func makeVirtualServer() v1.VirtualServer {
+	return v1.VirtualServer{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      "cafe",
+			Namespace: "default",
+		},
+		Spec: v1.VirtualServerSpec{
+			Host: "example.com",
+			TLS: &v1.TLS{
+				Secret: "abc",
+			},
+			ExternalDNS: v1.ExternalDNS{
+				Enable: false,
+			},
+			Upstreams: []v1.Upstream{
+				{
+					Name:      "first",
+					Service:   "service-1",
+					LBMethod:  "random",
+					Port:      80,
+					MaxFails:  createPointerFromInt(8),
+					MaxConns:  createPointerFromInt(16),
+					Keepalive: createPointerFromInt(32),
+					Type:      "grpc",
+				},
+				{
+					Name:    "second",
+					Service: "service-2",
+					Port:    80,
+				},
+			},
+			Routes: []v1.Route{
+				{
+					Path: "/first",
+					Action: &v1.Action{
+						Pass: "first",
+					},
+				},
+				{
+					Path: "/second",
+					Action: &v1.Action{
+						Pass: "second",
+					},
+				},
+			},
+			Dos: "some-ns/some-name",
+		},
+	}
+}
+
+func createPointerFromString(s string) *string {
+	return &s
+}
+
+func TestValidateFailsOnMissingBackupPort(t *testing.T) {
+	t.Parallel()
+
+	vs := makeVirtualServer()
+	// setup only backup name, missing backup port
+	vs.Spec.Upstreams[1].Backup = createPointerFromString("backup-service")
+
+	vsv := &VirtualServerValidator{isPlus: true, isDosEnabled: true}
+	err := vsv.ValidateVirtualServer(&vs)
+	if err == nil {
+		t.Error("want error for missing backup port, got nil", err)
+	}
+}
+
+func createPointerFromUInt16(port uint16) *uint16 {
+	return &port
+}
+
+func TestValidateFailsOnMissingBackupName(t *testing.T) {
+	t.Parallel()
+
+	vs := makeVirtualServer()
+	// setup only backup port, missing backup name
+	vs.Spec.Upstreams[1].BackupPort = createPointerFromUInt16(8080)
+
+	vsv := &VirtualServerValidator{isPlus: true, isDosEnabled: true}
+	err := vsv.ValidateVirtualServer(&vs)
+	if err == nil {
+		t.Error("want error for missing backup name, got nil", err)
+	}
+}
+
+func TestValidateFailsOnNotSupportedLBMethodForBackup(t *testing.T) {
+	t.Parallel()
+
+	notSupportedLBMethods := []string{"hash", "hash_ip", "random"}
```

Comment:
The default load balancing method we use is "random two least_conn". There was initial discussion around changing the default load balancing method if this setting is used so that the user isn't just hit with an error straight away: https://github.com/nginxinc/kubernetes-ingress/discussions/4091#discussioncomment-7104737

Reply:
good catch!